File: Mar2026/18thMarch2026/longestSubString.py
def longest_sub_string(s):
    left =0
    maxlength =1
    newSet = set()
    for right in range(len(s)):
        while s[right] in newSet :
            newSet.remove(s[left])
            left+=1
        
        newSet.add(s[right])

        if(right -left + 1 > maxlength):
            maxlength = right - left + 1

    return maxlength

print(longest_sub_string("abcabcbb"))

File: April2026/29thApril2026/longestSubstring.js
function longestSubstring(str) {
  let set = new Set();
  let result = "";
  let maxLength = 1;
  let left = 0;
  for (let right = 0; right < str.length; right++) {
    while (set.has(str[right])) {
      set.delete(str[left]);
      left++;
    }
    set.add(str[right]);
    if (right - left + 1 > maxLength) {
      maxLength = right - left + 1;
      result = str.slice(left, right + 1);
    }
  }
  return result
}


console.log(longestSubstring("abcadiwbeois"))
